tests/TestCase/Error/ExceptionRendererTest.php

```diff
@@ -731,23 +731,23 @@ public function testRenderExceptionInBeforeRender()
      */
     public function testMissingLayoutPathRenderSafe()
     {
+        $eventTriggered = false;
         $exception = new NotFoundException();
         $ExceptionRenderer = new ExceptionRenderer($exception);
 
-        $ExceptionRenderer->controller = $this->getMockBuilder('Cake\Controller\Controller')
-            ->setMethods(['render'])
-            ->getMock();
+        $ExceptionRenderer->controller = new Controller();
         $ExceptionRenderer->controller->helpers = ['Fail', 'Boom'];
-        $ExceptionRenderer->controller->eventManager()->on('Controller.beforeRender', function (Event $event) {
-            $event->subject()->viewBuilder()->layoutPath('boom');
-        });
+        $ExceptionRenderer->controller->eventManager()->on(
+            'Controller.beforeRender',
+            function (Event $event) use ($exception, &$eventTriggered) {
+                $eventTriggered = true;
+                $event->subject()->viewBuilder()->layoutPath('boom');
+
+                throw $exception;
```

Comment:
You're right, throwing the exception here is a little overkill. I did this in order to have the very same behaviour as before: the old mocked method `Controller::render()` did throw this exception.

However, I tried removing it and nothing changes, as far as I could see. I'll commit in a minute.